tempest/src/main/kotlin/app/cash/tempest/Query.kt

```diff
@@ -27,33 +27,91 @@ interface Queryable<K : Any, I : Any> {
    */
   fun query(
     keyCondition: KeyCondition<K>,
-    consistentRead: Boolean = false,
     asc: Boolean = true,
     pageSize: Int = 100,
+    consistentRead: Boolean = false,
     returnConsumedCapacity: ReturnConsumedCapacity = NONE,
     filterExpression: FilterExpression? = null,
     initialOffset: Offset<K>? = null
   ): Page<K, I>
+
+  // Overloaded functions for Java callers (Kotlin interfaces do not support `@JvmOverloads`).
+
+  fun query(keyCondition: KeyCondition<K>) = query(
+    keyCondition,
+    QueryConfig.Builder<K>().build()
+  )
+
+  fun query(keyCondition: KeyCondition<K>, config: QueryConfig<K>) = query(
+    keyCondition,
+    config.asc,
+    config.pageSize,
+    config.consistentRead,
+    config.returnConsumedCapacity,
+    config.filterExpression,
+    config.initialOffset
+  )
+}
+
+data class QueryConfig<K> internal constructor(
+  val asc: Boolean,
+  val pageSize: Int,
+  val consistentRead: Boolean,
+  val returnConsumedCapacity: ReturnConsumedCapacity,
+  val filterExpression: FilterExpression?,
+  val initialOffset: Offset<K>? = null
+) {
+  class Builder<K> {
+    private var asc: Boolean? = null
+    private var pageSize: Int? = null
```

Comment:
should this be non-null and default to 100 ? You can’t clear it, and that’s probably okay

Reply:
(similarly for all the parameters)